Delete FindFlatBuffers.cmake

Phatcat opened the pull request:
Use the provided flatbuffers-config.cmake

Changes:

- removed both findflatbuffer.cmake and buildflatbuffers.cmake

add_dependencies() does not actually make the module depend on it, it just ensures it's compiled before the module. It needs to be linked, and when it's linked it automatically becomes a dependency as well;

- Link to FB_SCHEMA_COMPILE where FB_SCHEMA_COMPILE used to be a dependency.

Small side-note: 
target_include_directories(${LIBRARY_NAME} PUBLIC ${CMAKE_BINARY_DIR})

is actually not for flatbuffers, it's for rpcgen, but it needs to be amended now since the old FindFlatBuffers propagated the CMake_Binary folder to the entirety of the project, effectively hiding (or removing, it's how you wanna look at it) any and all linking issues in regards to the binaries. Basically everything compiled to a daemon needed explicit linking to the binaries.

Chaosvex commented:
Looks good but can you restore the minimum required version?

Phatcat commented:
It would be cosmetic. It does not use it...

Edit*: This is not pcre this is flatbuffers, yeah it can check for a version. Of course.

Chaosvex commented:
That doesn't seem to be correct.

https://github.com/google/flatbuffers/blob/master/CMake/flatbuffers-config-version.cmake.in

Phatcat commented:
No you're right, I got confused and mixed it up with pcre, it does indeed check for version.
Pretty sure both mysql and botan is able to as well with their configs. Probably need to set minimum in mysql since the include structure changed in 8.0 and they're incompatible.

Phatcat committed:
Delete FindFlatBuffers.cmake

Use the provided flatbuffers-config.cmake and provided buildflatbuffers solution.
